### ISSUE TITLE
github.CommitFile is missing URL fields

### ISSUE THREAD

**Issue by bored-engineer** [google/go-github#419]
The `github.CommitFile` struct is missing a few fields. For example if we look at (shortened) file struct from [api.github.com/repos/google/go-github/commits/b9698164669e386680621e2006cab02271257611](https://api.github.com/repos/google/go-github/commits/b9698164669e386680621e2006cab02271257611):

``` json
{
  "sha": "916a2dc2a1a4699bf677505e0caad928ac5e0b6d",
  "filename": "github/search.go",
  "status": "modified",
  "additions": 12,
  "deletions": 8,
  "changes": 20,
  "blob_url": "https://github.com/google/go-github/blob/b9698164669e386680621e2006cab02271257611/github/search.go",
  "raw_url": "https://github.com/google/go-github/raw/b9698164669e386680621e2006cab02271257611/github/search.go",
  "contents_url": "https://api.github.com/repos/google/go-github/contents/github/search.go?ref=b9698164669e386680621e2006cab02271257611",
  "patch": "..."
},
```

Notice the `blob_url`, `raw_url`, and `contents_url` have no matching struct field in `github.CommitFile`. This is particularly evident when no `patch` value is returned (for example on binary files) and the file contents must be retrieved by the client (which requires the `contents_url`)

I didn't want to just submit a pull request adding the fields as there seems to be a pattern within `go-github` to avoid raw URLs and wanted to double check all the missing fields should be added or only some before attaching a pull request.

**Comment by willnorris** [comment]
Early in the development of the library, I was intentionally avoiding those raw URL fields since they are mostly useful if you are relying on the [hateoas](https://en.wikipedia.org/wiki/HATEOAS) design of the GitHub API.  go-github does not, so most of the time these URLs aren't needed.  There have been a few cases where people have needed them for various purposes, so I'm fine with them being added at this point.

**Comment by shurcooL** [comment]
In pursuit of closing non-actionable issues, now that #484 has been merged, is there anything else left to be done for this issue? Or shall we mark it as a question and close?

**Comment by bored-engineer** [comment]
Sorry for the delay. I believe this has been fixed by #484!
